prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -45,9 +47,13 @@ class DeliveryFunction(
      * @return Name for the organization
      */
     override fun getOrgName(organization: String): String? {
-        val receiver = workflowEngine.settings.findReceiver(organization)
-        receivingOrgSvc = receiver?.name
-        return receiver?.organizationName
+        return if (organization.contains(Sender.fullNameSeparator)) {
+            workflowEngine.settings.findReceiver(organization).also { receivingOrgSvc = it?.name }
+        } else {
+            workflowEngine.settings.receivers.find {
+                it.organizationName == organization && it.customerStatus == CustomerStatus.ACTIVE
+            }
+        }?.organizationName
```

Comment:
If a fully qualified `org.Receiver` string was supplied, find that receiver's org name & store its name for further filtering. Otherwise, find an active receiver for that org to validate the org name, but don't store the receiver name.

**We may want to change the filtering here (or get last instance instead of first)**